android/src/ti/map/PolylineProxy.java

```diff
@@ -0,0 +1,234 @@
+package ti.map;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.appcelerator.kroll.KrollProxy;
+import org.appcelerator.kroll.annotations.Kroll;
+import org.appcelerator.kroll.common.AsyncResult;
+import org.appcelerator.kroll.common.TiMessenger;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.TiContext;
+import org.appcelerator.titanium.util.TiConvert;
+
+import ti.map.Shape.IShape;
+import android.os.Message;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Polyline;
+import com.google.android.gms.maps.model.PolylineOptions;
+
+
+@Kroll.proxy(name="Polyline",creatableInModule=MapModule.class, propertyAccessors = {
+
+	MapModule.PROPERTY_STROKE_COLOR, MapModule.PROPERTY_STROKE_WIDTH,
+
+	PolylineProxy.PROPERTY_STROKE_COLOR2,
+	PolylineProxy.PROPERTY_STROKE_WIDTH2,
+
+	PolylineProxy.PROPERTY_ZINDEX,
+
+	MapModule.PROPERTY_POINTS,
+
+})
+public class PolylineProxy extends KrollProxy implements IShape
+{
+
+	private PolylineOptions options;
+	private Polyline polyline;
+
+	private static final int MSG_FIRST_ID = KrollProxy.MSG_LAST_ID + 1;
+
+	private static final int MSG_SET_POINTS 	  = MSG_FIRST_ID + 499;
+	private static final int MSG_SET_STROKE_COLOR = MSG_FIRST_ID + 501;
+	private static final int MSG_SET_STROKE_WIDTH = MSG_FIRST_ID + 502;
+	private static final int MSG_SET_ZINDEX 	  = MSG_FIRST_ID + 503;
+
+	public static final String PROPERTY_STROKE_COLOR2 = "color";
+	public static final String PROPERTY_STROKE_WIDTH2 = "width";
+
+	public static final String PROPERTY_ZINDEX = "zIndex";
+
+	public PolylineProxy() {
+		super();
+	}
+
+	public PolylineProxy(TiContext tiContext) {
+		this();
+	}
+
+	@Override
+	public boolean handleMessage(Message msg)
+	{
+
+		//	MSG_SET_POINTS
+		//	MSG_SET_STROKE_WIDTH
+		//	MSG_SET_STROKE_COLOR
+		//	MSG_SET_FILL_COLOR
+		//	MSG_SET_ZINDEX
+		AsyncResult result = null;
+		switch (msg.what) {
+			case MSG_SET_POINTS: {
+				result = (AsyncResult) msg.obj;
+				polyline.setPoints(processPoints(result.getArg(), true));
+				result.setResult(null);
+				return true;
+			}
+			case MSG_SET_STROKE_WIDTH: {
+				result = (AsyncResult) msg.obj;
+				polyline.setWidth((Float)result.getArg());
+				result.setResult(null);
+				return true;
+			}
+			case MSG_SET_STROKE_COLOR: {
+				result = (AsyncResult) msg.obj;
+				polyline.setColor((Integer)result.getArg());
+				result.setResult(null);
+				return true;
+			}
+			case MSG_SET_ZINDEX: {
+				result = (AsyncResult) msg.obj;
+				polyline.setZIndex((Float)result.getArg());
+				result.setResult(null);
+				return true;
+			}
+			default : {
+				return super.handleMessage(msg);
+			}
+		}
+	}
+	public void processOptions() {
+
+		options = new PolylineOptions();
+		String op;
```

Comment:
Same as previous part:
Is the String op needed? Would it be possible to do the processOptions() method without it?